HearthDb/Card.cs

```diff
@@ -116,18 +129,38 @@ public string GetLocText(Locale lang)
 			var text = Entity.GetLocString(CARDTEXT_INHAND, lang)?.Replace("_", "\u00A0").Trim();
 			if(text == null)
 				return null;
-			var index = text.IndexOf('@');
-			if (index == 0)
+			var count = text.Contains("@");
+
+			if (!count && !text.Contains("|4"))
 				return text;
-			if (Entity.GetTag(PLAYER_TAG_THRESHOLD_TAG_ID) > 0)
-				return text.Substring(0, index);
-			var scriptData1 = Entity.GetTag(TAG_SCRIPT_DATA_NUM_1);
-			if (scriptData1 > 0)
-				return text.Replace("@", scriptData1.ToString());
-			var scriptData2 = Entity.GetTag(TAG_SCRIPT_DATA_NUM_2);
-			if (scriptData2 > 0)
-				return text.Replace("@", scriptData2.ToString());
-			return text.Substring(index + 1);
+
+			int? num = null;
+
+			if (SpellstoneStrings.ContainsKey(Entity.CardId))
+				return text.Replace("@", "");
+
+			if (Entity.GetTag(TAG_SCRIPT_DATA_NUM_1) != 0)
+				num = Entity.GetTag(TAG_SCRIPT_DATA_NUM_1);
+			else if (Entity.GetTag(SCORE_VALUE_1) != 0)
+				num = Entity.GetTag(SCORE_VALUE_1);
+			if (num != null)
+				text = text.Replace("@", num.ToString());
+
+			var atCounterMatch = atCounterRegex.Matches(text);
+			if (atCounterMatch.Count > 0)
+            {
+				var replacement = num == 1 ? atCounterMatch[0] : atCounterMatch[1];
+				text = text.Substring(0, text.IndexOf(atCounterMatch[0].ToString())) + replacement.ToString() + text.Substring(text.IndexOf(atCounterMatch[1].ToString()), text.Length - 1);
```

Comment:
You can omit ` text.Length - 1`: `... + text.Substring(text.IndexOf(atCounterMatch[1].ToString())`